### DISCUSSION TITLE
Inclusive Features to Phet sims

### DISCUSSION THREAD

**Discussion by gnaqi** [scenerystack/community#46] [Q&A] [answered]
Hi all
I'm a very new to phet sims development. I have a question, is it possible to add inclusive features (in Arabic) like interactive description to an available sim (like "pH Scale: Basics") ? If yes, where to start?

**Comment by brettfiedler** [comment, marked as answer]
Hi @gnaqi!

It is wonderful to hear that you'd like to enhance the accessibility for the simulations, in Arabic too!

It is indeed possible, though the size of the project depends on your starting point. I'll list out some recommended steps and resources for different paths.

If you're already familiar with web development (HTML, JavaScript/TypeScript), then you may only need to spend some time understanding the sim development process and sim code architecture, then looking at some sim examples with Interactive Description and then diving into some documentation

## Getting Familiar with SceneryStack code and Simulation Development

1. Set up and get familiar with a simulation development environment by following along with the [PhET Development Overview](https://scenerystack.github.io/community/info-sync/simulation-development-overview/)
2. Read through the [Scenery Foundational Documentation](https://scenerystack.github.io/community/guides/scenery/#get-to-know-scenery) for an understanding of the scene framework, which will help you to understand how accessible components are integrated into it.

### Getting Familiar with Scenery Accessibility and Inclusive Design Resources

1. Read through the [Scenery Accessibility Documentation](https://phetsims.github.io/scenery/doc/accessibility/accessibility.html)
2. Pick a [few simulations with Interactive Description](https://phet.colorado.edu/en/simulations/filter?type=html&a11yFeatures=interactive-description) and look through their code. You'll find the source code for all simulations on the [phetsims GitHub page](https://github.com/phetsims/) (just search for the simulation you're interested in and peruse the code!)
3. Interactive Description will require [Alternative Input](https://scenerystack.github.io/community/info-sync/alternative-input-quickstart-guide/), which will enable and enhance inputs such as keyboards and switches.
4. Then it's time to follow along with the [Technical Guide for adding Interactive Description](https://scenerystack.github.io/community/info-sync/interactive-description-technical-guide/).

At this stage, you may find it helpful to try your hand at modifying a simulation with existing interactive description and translating the strings into Arabic, before building interactive description into a simulation from scratch.

You can also find more [resources for inclusive features and SceneryStack Accessibility](https://scenerystack.github.io/community/accessibility/phet_a11y_guides/), including an entire Coursera course for designing descriptions within the Interactive Description framework.

Here is the source code for the simulation, pH Scale: Basics, that you indicated above: https://github.com/phetsims/ph-scale-basics

## Little Experience? Getting Started with Web Development, SceneryStack, and Simulation Development

If you're starting from a place with little or no web development experience, you may find [the Development Pathways](https://scenerystack.github.io/community/guides/pathways/) page a good jumping off point. From there, when you're comfortable, you can tackle the resources shared above. If you have access to any chat-based large AI models, I do recommend using it to help you understand any code you see! Just copy and paste the code in and ask about any part.

No matter what path you choose, please continue to share your thoughts, questions, and progress! ❤️ We're in the beginning stages of growing our developer community, and we're thrilled to have you as a part. Feel free to join our [Slack or Matrix channel](https://scenerystack.github.io/community/join/), if you have not already and share your progress there (or here!).

**Reply by gnaqi** [reply]
Hi @brettfiedler 
Thank you for your prompt and comprehensive reply. Yes I have good experience in web development, HTML and Javascript. 
I will go through the steps you mentioned and get back to you if I need more details.
Thank you

**Comment by terracoda** [comment]
Hi @gnaqi, I also think it is wonderful that you are interested in the Interactive Description feature. @brettfiedler has kindly provided links to resources that help with development and implementation of the feature, which is itself a lengthy process. I am curious if your question is also about  how to design the actual descriptions that would be needed to create a good learner experience for pH Scale:Basics?

Is your end goal, access to description for screen reader users, or access to descriptions for everyone, or perhaps both?

The reason I ask is because the description design itself requires particular expertise, and we have additional resources about description design as well.

**Reply by terracoda** [reply]
Via the SceneryStack website, @brettfiedler did indeed link to the [Description Design for Interactive Learning Resources](https://www.coursera.org/learn/description-design-for-interactive-learning-resources). This Coursera course (our course) dives into our Description Design Framework and the description design process. The framework is for designing descriptions whether it be for one of our features, Interactive Description or Voicing, or some other interactive context.

**Reply by gnaqi** [reply]
Hi @terracoda 
Our goal is to make the sims accessible for blind students (implementing the feature of using keyboard as input and voice descriptions in Arabic). One of our team members are Dr. Dduha Chehada she is more in concern with the description design itself. For me I'm responsible more of the implementation. Of course I will check the training course suggested I'm sure it will be helpful.
